mirrorlist.go

```diff
@@ -10,51 +10,45 @@ import (
 )
 
 func updateMirrorlists() {
-	for name, _ := range config.Repos {
-		err := checkAndUpdateMirrorlistRepo(name)
+	for name, repo := range config.Repos {
+		err := checkAndUpdateMirrorlistRepo(name, repo)
 		if err != nil {
 			log.Fatal(err)
 		}
 	}
 }
 
-func checkAndUpdateMirrorlistRepo(repoName string) error {
-	repo, ok := config.Repos[repoName]
-	if !ok {
-		return fmt.Errorf("repo %v does not exist in config", repoName)
-	}
+func checkAndUpdateMirrorlistRepo(repoName string, repo *Repo) error {
 	if repo.Mirrorlist != "" {
-		lastCheck, ok := lastMirrorlistCheck[repo.Mirrorlist]
-		if ok && time.Since(lastCheck) < 5*time.Second {
+		repo.mutex.Lock()
+		defer repo.mutex.Unlock()
+
+		if time.Since(repo.LastMirrorlistCheck) < 5*time.Second {
 			// if there is an entry in the lastMirrorlistCheck and that entry has a distance lower than 5 seconds from now, don't update its mirrorlist
 			return nil
 		}
-		lastMirrorlistCheck[repo.Mirrorlist] = time.Now()
-		err := updateRepoMirrorlist(repoName)
+
+		repo.LastMirrorlistCheck = time.Now()
+		err := updateRepoMirrorlist(repoName, repo)
 		if err != nil {
 			return fmt.Errorf("error while updating %v repo mirrorlist: %v", repoName, err)
 		}
 	}
 	return nil
 }
 
-func updateRepoMirrorlist(repoName string) error {
-	repo, ok := config.Repos[repoName]
-	if !ok {
-		return fmt.Errorf("repo %v does not exist in config", repoName)
-	}
+func updateRepoMirrorlist(repoName string, repo *Repo) error {
```

Comment:
locking in this function or on the caller? I have no clue on what is the proper option

Reply:
It makes sense to me! I had separated them when I wrote them because I didn't knew how the code would have evolved